cmd/kubeadm/app/phases/addons/dns/dns_test.go

```diff
@@ -1383,6 +1383,75 @@ func TestCreateDNSService(t *testing.T) {
 	}
 }
 
+func TestDeployedDNSAddon(t *testing.T) {
+	newMockClientWithImage := func(t *testing.T, img string) *clientsetfake.Clientset {
+		ctx := context.Background()
+		t.Helper()
+		client := newMockClientForTest(t, 1, 1)
+		deploy, err := client.AppsV1().Deployments(metav1.NamespaceSystem).Get(ctx, "coredns-0", metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("error getting CoreDNS deployment: %v", err)
+		}
+		deploy.Spec.Template.Spec.Containers = []v1.Container{{Image: img}}
+		if _, err := client.AppsV1().Deployments(metav1.NamespaceSystem).Update(ctx, deploy, metav1.UpdateOptions{}); err != nil {
+			t.Fatalf("error updating CoreDNS deployment: %v", err)
+		}
+		return client
```

Comment:
Instead of updating the deployment image with many API calls, how about just adding a new parameter for `newMockClientForTest` helper function, for example :
```
func newMockClientForTest(t *testing.T, replicas int32, deploymentSize int, image string) *clientsetfake.Clientset {
```
That seems more straightforward

Reply:
Thanks for the suggestion, implemented.